pkg/resources/materialized_view.go

```diff
@@ -0,0 +1,316 @@
+package resources
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/csv"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+var materializedViewSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the identifier for the view; must be unique for the schema in which the view is created.",
+	},
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database in which to create the view. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The schema in which to create the view. Don't use the | character.",
+		ForceNew:    true,
+	},
+	"or_replace": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Overwrites the View if it exists.",
+	},
+	"is_secure": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Specifies that the view is secure.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the view.",
+	},
+	"statement": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      "Specifies the query used to create the view.",
+		ForceNew:         true,
+		DiffSuppressFunc: DiffSuppressStatement,
+	},
```

Comment:
I think we want to make `warehouse` required here

Reply:
ok will do

Reply:
@alldoami No place to specify warehouse in materialized view. https://docs.snowflake.com/en/sql-reference/sql/create-materialized-view.html